NitroxClient/MonoBehaviours/Gui/InGame/Modal.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UWE;
+
+namespace NitroxClient.MonoBehaviours.Gui.InGame;
+
+/// <summary>
+/// Base class for Modal components, which are dialog boxes that appear in the middle of the screen
+/// </summary>
+public abstract class Modal : MonoBehaviour
```

Comment:
Except from `StartCoroutine()`  in L76, which can be replaced by `CoroutineHost.StartCoroutineAsync()`, I don't see a reason this is inheriting MonoBehaviour.

Reply:
We used to add the modals as components in the Multiplayer GameObject. Do you think we don't need to add them as components anymore ? (since we now have a static dictionary with their references)

Reply:
Yes as we don't need Start(), Update, etc. we can (and should) remove the inheritance of MonoBehaviour.